master/internal/api_project.go

```diff
@@ -439,10 +439,12 @@ func (a *apiServer) deleteProject(ctx context.Context, projectID int32,
 	}
 
 	log.Debugf("deleting project %d experiments", projectID)
-	if _, err = a.deleteExperiments(expList, user); err != nil {
-		log.WithError(err).Errorf("failed to delete experiments")
-		_ = a.m.db.QueryProto("delete_fail_project", holder, projectID, err.Error())
-		return err
+	if len(expList) > 0 {
```

Comment:
suggestion: move both log statements in the conditional

Reply:
also deleteExperiments looks like it should be handling an empty expList gracefully 🤔 what's the error msg you saw?

Reply:
This was the error message: Failed to delete workspace: failed to gather trial IDs for experiment: querying for trial IDs of experiments []: ERROR: syntax error at or near ")" (SQLSTATE 42601)

Reply:
that looks like is coming from https://github.com/determined-ai/determined/blob/46379a63a2b536c13a6c5a8e685c7476250a633c/master/internal/db/postgres_experiments.go#L1048 what do you think about fixing it there? I expected bun and postgres to handle this better 🤔

Reply:
Just took a quick peek, looks like that error is from `deleteExperiments()` line 440. I would also prefer moving the length check inside `deleteExperiments`, perhaps after line 435 in that fn?

Reply:
or I think it just shouldn't error. if it's given an empty list of experiments it gives back an empty list of task ids etc

Reply:
I agree with @eecsliu about moving the length check inside `deleteExperiments` after line 435 because after the `ExperimentsTrialAndTaskIDs` function [this](https://github.com/determined-ai/determined/blob/46379a63a2b536c13a6c5a8e685c7476250a633c/master/internal/db/postgres_experiments.go#L885) function is called which will also fail when `expIDs` is empty. I can fix `ExperimentsTrialAndTaskIDs` to return an empty list of task ids and also have the length check in `deleteExperiments`

Reply:
that works. although IMO if https://github.com/determined-ai/determined/blob/46379a63a2b536c13a6c5a8e685c7476250a633c/master/internal/db/postgres_experiments.go#L885 also doesn't fail when given an empty list of experiments we'll be better off. unless we think erroring in these cases communicates a bug and we don't want to lose that